pycbc/events/stat.py

```diff
@@ -2083,6 +2083,83 @@ def lognoiserate(self, trigs):
         logr_n += self.find_dq_val(trigs)
         return logr_n
 
+class DQExpFitFgBgKDEStatistic(DQExpFitFgBgNormStatistic):
+    """
+    The ExpFitFgBgKDEStatistic with DQ-based reranking.
+
+    This is the same as the DQExpFitFgBgNormStatistic except the signal
+    rate is adjusted according to the KDE statistic files
+    """
+
+    def __init__(self, sngl_ranking, files=None, ifos=None, **kwargs):
+        """
+        Parameters
+        ----------
+        sngl_ranking: str
+            The name of the ranking to use for the single-detector triggers.
+        files: list of strs, needed here
+            A list containing the filenames of hdf format files used to help
+            construct the coincident statistics. The files must have a 'stat'
+            attribute which is used to associate them with the appropriate
+            statistic class.
+        ifos: list of strs, not used here
+            The list of detector names
+        """
+        DQExpFitFgBgNormStatistic.__init__(self, sngl_ranking, files=files,
+                                           ifos=ifos, **kwargs)
+        # The stat file attributes are hard-coded as 'signal-kde_file'
```

Comment:
I expect codeclimate complained about this, and (IMO) correctly - these lines should rather be refactored as another method of the ExpFitFgBgKDEStatistic (eg `find_kdes` ?).  Hopefully this also won't require you to define `assign_kdes` redundantly here.